lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -414,35 +414,40 @@ private EventHandler getEventHandler(final String storageCode)
 
     private void processEvent(final StatusEvent ev)
             throws IndexingException, InterruptedException, RetriableIndexingException {
+
+        // update event to reflect the latest state of object for which the specified StatusEvent is an event for
+        EventHandler handler = getEventHandler(ev.getStorageCode());
+        StatusEvent updatedEvent = handler.updateObjectEvent(ev);
+
         try {
-            switch (ev.getEventType()) {
+            switch (updatedEvent.getEventType()) {
             case NEW_VERSION:
-                GUID pguid = ev.toGUID();
+                GUID pguid = updatedEvent.toGUID();
                 boolean indexed = indexingStorage.checkParentGuidsExist(new LinkedHashSet<>(
                         Arrays.asList(pguid))).get(pguid);
                 if (indexed) {
                     logger.logInfo("[Indexer]   skipping " + pguid +
                             " creation (already indexed)");
                     // TODO: we should fix public access for all sub-objects too (maybe already works. Anyway, ensure all subobjects are set correctly as well as the parent)
-                    if (ev.isPublic().get()) {
+                    if (updatedEvent.isPublic().get()) {
                         publish(pguid);
                     } else {
                         unpublish(pguid);
                     }
                 } else {
-                    indexObject(pguid, ev.getStorageObjectType().get(), ev.getTimestamp(),
-                            ev.isPublic().get(), null, new LinkedList<>());
+                    indexObject(pguid, updatedEvent.getStorageObjectType().get(), updatedEvent.getTimestamp(),
+                            updatedEvent.isPublic().get(), null, new LinkedList<>());
                 }
                 break;
             // currently unused
 //            case DELETED:
 //                unshare(ev.toGUID(), ev.getAccessGroupId().get());
 //                break;
             case DELETE_ALL_VERSIONS:
```

Comment:
According to codecov all the lines with changes below this line are untested

Reply:
Looks like it is covered. This line is covered in my local test report.

Reply:
I meant all the lines below the currentl line with changes in them, not the current line.

Reply:
I'm talking about the lines below this line, not this line in particular. It looks like variable names changes mostly.

Reply:
This needs to be a separate ticket/pr since these tests are unrelated to the event update function. This code requires tests for rename, publish, unpublish functions.